src/ESM.sol

```diff
@@ -21,31 +21,37 @@ pragma solidity >=0.6.12;
 
 interface GemLike {
     function balanceOf(address) external view returns (uint256);
+    function burn(uint256) external;
     function transfer(address, uint256) external returns (bool);
     function transferFrom(address, address, uint256) external returns (bool);
 }
 
 interface EndLike {
+    function live() external view returns (uint256);
+    function vat()  external view returns (address);
     function cage() external;
 }
 
+interface VatLike {
+    function deny(address) external;
+}
+
 contract ESM {
-    GemLike public immutable gem; // collateral
-    EndLike public immutable end; // cage module
-    address public immutable pit; // burner
-    uint256 public immutable min; // threshold
-    uint256 public fired;
+    GemLike public immutable gem;   // collateral
+    EndLike public immutable end;   // cage module
+    address public immutable proxy; // Pause proxy
+    uint256 public immutable min;   // threshold
 
     mapping(address => uint256) public sum; // per-address balance
     uint256 public Sum; // total balance
 
     event Fire();
     event Join(address indexed usr, uint256 wad);
 
-    constructor(address gem_, address end_, address pit_, uint256 min_) public {
+    constructor(address gem_, address end_, address proxy_, uint256 min_) public {
         gem = GemLike(gem_);
         end = EndLike(end_);
-        pit = pit_;
+        proxy = proxy_;
```

Comment:
The comment on `proxy` helps, but I'd suggest a different or more descriptive name here. I briefly entertained the idea that we were storing MKR in a ds-proxy.

Reply:
My original idea was using the full `pauseProxy` but it looked huge vs the other variables.
But if we consider that is the best way, not opposed to set it back to that.